content/usgs-packages/sbtools_Discovery.Rmd

```diff
@@ -0,0 +1,224 @@
+head(precip_maps_data, 2)
+
+##### -------------------------------------- #####
+
+# search by 2 keywords
+hazard2_query <- list(q = 'flood', q = 'earthquake')
+hazard2_data <- query_sb(query_list = hazard2_query)
+length(hazard2_data)
+head(hazard2_data, 2)
+
+# search by 3 keywords
+hazard3_query <- list(q = 'flood', q = 'earthquake', q='drought')
+hazard3_data <- query_sb(query_list = hazard3_query)
+length(hazard3_data)
+head(hazard3_data, 2)
+```
+
+### Using `query_sb_text`
+
+`query_sb_text` returns a list of `sbitems` that match the title or description fields. Use it to search authors, station names, rivers, states, etc.
+
+```{r}
+# search using a contributors name
+contrib_results <- query_sb_text("Robert Hirsch")
+head(contrib_results, 2)
+
+# search using place of interest
+park_results <- query_sb_text("Yellowstone")
+head(park_results, 2)
+
+# search using a site location
+loc_results <- query_sb_text("Embudo")
+length(loc_results)
+head(loc_results, 2)
+```
+
+### Using `query_sb_doi`
+
+Use a Digital Object Identifier (DOI) to query ScienceBase. This should return only one list item, unless there is more than one ScienceBase item referencing this very unique identifier.
+
+```{r}
+# USGS Microplastics study 
+query_sb_doi('10.5066/F7ZC80ZP')
+
+####### I've tried A TON of DOIs I found through the web interface and just
+####### keep getting empty lists returned.
+query_sb_doi('10.1016/j.coldregions.2007.05.009')
+```
+
+### Using `query_sb_spatial`
+
+`query_sb_spatial` accepts 3 different methods for specifying a spatial area in which to look for data. You can specify a bounding box `bbox` as an `sp` spatial data object *[[[[NEED MORE TEXT HERE]]]]*.  Alternatively, you can supply a vector of latitudes and a vector of longitudes using `lat` and `long` arguments. The function will automatically use the minimum and maximum from those vectors to construct a boundary box. The last way to represent a spatial region to query ScienceBase is using a POLYGON Well-known text (WKT) object as a text string. The format is `"POLYGON(([LONG1 LAT1], [LONG2 LAT2], [LONG3 LAT3]))"`, where `LONG#` and `LAT#` are longitude and latitude pairs as decimals. See the [Open Geospatial Consortium WKT standard](http://www.opengeospatial.org/standards/wkt-crs) for more information. 
+
+```{r}
+### SPATIAL QUERY EXAMPLES NEED SOME TLC
+
+appalachia <- data.frame(
+  lat = c(34.576900, 36.114974, 37.374456, 35.919619, 39.206481),
+  long = c(-84.771119, -83.393990, -81.256731, -81.492395, -78.417345))
+
+conus <- data.frame(
+  lat = c(49.078148, 47.575022, 32.914614, 25.000481),
+  long = c(-124.722111, -67.996898, -118.270335, -80.125804))
+
+# verifying where points are supposed to be
+maps::map('usa')
+points(conus$long, conus$lat, col="red", pch=20)
+points(appalachia$long, appalachia$lat, col="green", pch=20)
+
+# query by bounding box
+bbox_sp_obj <- sp::SpatialPoints(appalachia)
+# query_sb_spatial(bbox=bbox_sp_obj)
```

Comment:
I ran a query by interactively selecting a bounding box under "Browse by Location" at https://www.sciencebase.gov/catalog/. The resulting URL was
```
https://www.sciencebase.gov/catalog/items?filter=spatialQuery%3D%7Btype%3A%22envelope%22%2Ccoordinates%3A%5B%5B-91.06475439079985%2C39.171001192353756%5D%2C+%5B-81.22100439080239%2C33.21851920313964%5D%5D%7D
```
which shares the `spatialQuery` filter with `query_sb_spatial`, but otherwise looks different from the WKT format that's ultimately used by all calls to `query_sb_spatial` (https://github.com/USGS-R/sbtools/blob/master/R/query_sb_spatial.R#L59). do you suppose SB might have changed formats? i don't see reference to WKT in their spatial docs.